src/diagnostic-channel-publishers/src/pg.pub.ts

```diff
@@ -0,0 +1,160 @@
+                        queryResult.emit("error", err);
+                    }
+                } else if (cb) {
+                    cb.apply(this, arguments);
+                }
+            });
+
+            try {
+                Object.defineProperty(trackingCallback, diagnosticOriginalFunc, { value: cb });
+                return trackingCallback;
+            } catch (e) {
+                // this should never happen, but bailout in case it does
+                return cb;
+            }
+        }
+
+        // this function takes too many variations of arguments.
+        // this patches any provided callback or creates a new callback if one wasn't provided.
+        // since the callback is always called (if provided) in addition to always having a Promisified
+        // EventEmitter returned (well, sometimes -- see above), its safe to insert a callback if none was given
+        try {
+            if (typeof config === "string") {
+                if (values instanceof Array) {
+                    data.query.preparable = {
+                        text: config,
+                        args: values,
+                    };
+                    callback = patchCallback(callback);
+                } else {
+                    data.query.text = config;
+
+                    // pg v6 will, for some reason, accept both
+                    // client.query("...", undefined, () => {...})
+                    // **and**
+                    // client.query("...", () => {...});
+                    // Internally, precedence is given to the callback argument
+                    if (callback) {
+                        callback = patchCallback(callback);
+                    } else {
+                        values = patchCallback(values);
+                    }
+                }
+            } else {
+                if (typeof config.name === "string") {
+                    data.query.plan = config.name;
+                } else if (config.values instanceof Array) {
+                    data.query.preparable = {
+                        text: config.text,
+                        args: config.values,
+                    };
+                } else {
+                    data.query.text = config.text;
+                }
+
+                if (callback) {
+                    callback = patchCallback(callback);
+                } else if (values) {
+                    values = patchCallback(values);
+                } else {
+                    config.callback = patchCallback(config.callback);
+                }
+            }
+        } catch (e) {
+            // if our logic here throws, bail out and just let pg do its thing
+            return originalClientQuery.apply(this, arguments);
+        }
+
+        arguments[0] = config;
+        arguments[1] = values;
+        arguments[2] = callback;
+        arguments.length = (arguments.length > 3) ? arguments.length : 3;
```

Comment:
I can make another PR that adds this for all of the modules? This sort of arguments trickery is needed whenever we want to support calling with more than the named arguments _and_ fewer than the named arguments.